cpu/atmega1284p/Makefile.dep

```diff
@@ -0,0 +1,5 @@
+ifneq (,$(filter atmega_pcint,$(USEMODULE)))
+    USEMODULE += atmega_pcint0 atmega_pcint1 atmega_pcint2
```

Comment:
I'd put that just into the `Makefile.include` of each individual AVR chip and keep `boards/common/arduino-atmega/Makefile.dep` unchanged and drop this file. Also, the ATmega1284P does have PCINT3 in contrast to the ATmege328P and ATmega2560.

Reply:
I also guess that all AVR MCUs have at least PCINT0 to PCINT2, so maybe move the code above to  `cpu/atmega_common/Makefile.include` and just add

```
ifneq (,$(filter atmega_pcint,$(USEMODULE)))
    USEMODULE += atmega_pcint3
endif
```

to `cpu/atmega1284p/Makefile.include` to have only the special case handled at CPU level

Reply:
OK, my guess is right at least for the boards supported by RIOT.

ATmega1281, ATmega328P, ATmega2560, ATmega256RFR2: PCINT0 to PCINT2
ATmega1284P: PCINT0 to PCINT3

So I'd just add PCINT0 to PCINT2 for all AVRs and add special case code to the ATmega1284p

Reply:
Sounds good, will update. I will also check with .include and .dep. As far as I understand .dep should contain modules. Additionally, the order of Makefiles might matter because of FEAUTRES_REQUIRED/PROVIDED, I had some strange errors yesterday.

Reply:
You are right. Most code in `cpu` does not do this properly (unlike in `boards` or `drivers`, where this distinction is done consistently). Recently added code (e.g. the ESP32, ESP8266, and RISC-V) does comply with this. Maybe it is a good idea to follow their lead and also start using that distinction.

Reply:
Looking at other boards, I kept FEATURES_REQUIRED and USEMODULE expansion within Makefile.dep as it seems makes a lot more sense to me to have it in `Makefile.dep`.